cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserPermissionsManager.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2023 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.stream.Collectors;
+import java.util.Collections;
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import org.eclipse.jetty.security.UserStore;
+import org.eclipse.jetty.security.PropertyUserStore;
+import org.eclipse.jetty.security.AbstractLoginService;
+import com.linkedin.kafka.cruisecontrol.config.constants.WebServerConfig;
+import javax.security.auth.Subject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The util class for User Permissions
+ */
+public class UserPermissionsManager {
+    private static final Logger LOG = LoggerFactory.getLogger(UserPermissionsManager.class);
+
+    private final KafkaCruiseControlConfig _config;
+    private final Map<String, Set<String>> _rolesPerUsers;
+
+    public UserPermissionsManager(KafkaCruiseControlConfig config) {
+        _config = config;
+        _rolesPerUsers = createRolesPerUsersMap();
+    }
+
+    /** Builds a map between existing users and their roles in Cruise Control
+     *
+     * @return a map of usernames -> their assigned roles
+     */
+    private Map<String, Set<String>> createRolesPerUsersMap() {
+        Map<String, Set<String>> rolesPerUsers = new HashMap();
+        boolean securityEnabled = _config.getBoolean(WebServerConfig.WEBSERVER_SECURITY_ENABLE_CONFIG);
+        if (securityEnabled) {
+            String privilegedFilePath = _config.getString(WebServerConfig.WEBSERVER_AUTH_CREDENTIALS_FILE_CONFIG);
+            UserStore userStore = createUserStoreFromFile(privilegedFilePath);
+            startUserStore(userStore);
+
+            Set<String> userNames = userStore
+                    .getKnownUserIdentities()
+                    .keySet();
+
+            for (String user : userNames) {
+                Subject userSubject = userStore.getUserIdentity(user).getSubject();
+                Set<AbstractLoginService.RolePrincipal> roles = userSubject == null
```

Comment:
I'm not familiar with the auth library. Would it be normal if userSubject is null? If that's abnormal, we should log some warnings.

Reply:
If it is null it just means that it was not found in the userStore object created from the WebServerConfig.WEBSERVER_AUTH_CREDENTIALS_FILE_CONFIG file. So if this user does not exist in this file, it will be null and will have no roles == permissions to do anything.